File: javascript/series/series.js
export class Series {
    constructor(input) {
        this.digits = [...input].map(Number);
    }

    slices(len) {
        if (len > this.digits.length) {
            throw new Error('Slice size is too big.');
        }
        return Array(this.digits.length-len+1).fill(null)
            .map((_, i) => this.digits.slice(i, i+len))
    }
}

File: java/series/src/main/java/Series.java
import java.util.List;
import java.util.stream.IntStream;
import java.util.stream.Collectors;

class Series {
    private final String series;

    Series(String series) {
        this.series = series;
    }

    List<String> slices(int len) {
        if (len < 1) {
            throw new IllegalArgumentException("Slice size is too small.");
        }

        if (len > series.length()) {
            throw new IllegalArgumentException("Slice size is too big.");
        }

        return IntStream.range(0, series.length() - len + 1)
            .mapToObj(i -> series.substring(i, i + len))
            .collect(Collectors.toList());
    }
}
